src/RestaurantPage.jsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from 'react';
+
+import { useParams } from 'react-router-dom';
+
+import { useDispatch } from 'react-redux';
+
+import { loadRestaurant, setRestaurant } from './actions';
+
+import RestaurantContainer from './RestaurantContainer';
+
+export default function RestaurantPage() {
+  const dispatch = useDispatch();
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    dispatch(setRestaurant({ id }));
+    dispatch(loadRestaurant());
```

Comment:
```suggestion
    dispatch(loadRestaurant({ id }));
```

`loadRestaurant`을 할때 id 를 넘겨주면 어떤가요?

Reply:
원래는 loadRestaurant에 id ( restaurantId a.k.a query String ) 를 인자로 넘겨서 처리하는 방식을 고려해봤었는데, 

loadRestaurant에서 getState()를 활용하여, 현재 store의 상태를 참고 할 수 있게 했습니다. 

```
export function loadRestaurant() {
  return async (dispatch, getState) => {
    const {
      restaurant: {
        id: restaurantId,
      },
    } = getState(); <- 활용 부분 

    const restaurant = await fetchRestaurant({ restaurantId });

    dispatch(setRestaurant(restaurant));
  };
}
```


**"둘다 동작은 어차피 같은데, 그냥 넘겨도 되지 않음?"** 이라는 의문이 들 수 있다고 생각 할 수 있다고 합니다. 


앞서 언급 했듯이, loadRestaurant로 넘겨지는 인자에 대한 의존성을 낮추고, 
현재 갱신된 store의 state를 참조해서 활용하자라는게 제 의도였습니다. 

음... 이렇게 작성하게 되면 발생을 할 수 있는 문제가, 
store에 연결 된 component 중에 restaurant 를 주시하고 있다면, 두번 갱신되기 떄문에, 해당 컴포넌트의 특정 로직이 엉킬수도 있겠다고 생각이되네요. 

의존성을 낮춰주는 이점이 있는 반면, 에러를 발생시킬수도 있다는 단점도 있겠네요. 

tradeoff를 따지면... 후자가 영향이 더 크니 변경하는 것을 추천하실까요?